### PR TITLE
Labeled field updates

### PR DESCRIPTION
**Pull request by jhp0621**
Addresses: 
https://github.com/LaunchPadLab/lp-components/issues/436
https://github.com/LaunchPadLab/lp-components/issues/624

## Author Checklist

- [x] Add unit test(s)
- [x] Update version in package.json (see the [versioning guidelines](https://github.com/LaunchPadLab/opex-public/blob/main/gists/npm-package-guidelines.md#pull-requests-and-deployments))
- [x] Update documentation (if necessary)
- [x] Add story to storybook (if necessary)
- [x] Assign dev reviewer

### CODE REVIEW

File: src/forms/inputs/checkbox-group.js [hunk 1]
```diff
@@ -88,17 +91,45 @@ const propTypes = {
...
-  return <legend>{label || convertNameToLabel(name)}</legend>
+function CheckboxGroupLegend({
+  label,
+  name,
+  required,
+  requiredIndicator,
+  hint,
+}) {
+  return (
+    <legend className={classnames({ 'visually-hidden': label === false })}>
```
Comment (jhp0621) [inline src/forms/inputs/checkbox-group.js:112]: source: https://www.w3.org/TR/WCAG20-TECHS/H71.html#:~:text=The%20first%20element%20inside%20the,related%20radio%20buttons%20and%20checkboxes.
> The first element inside the fieldset must be a legend element, which provides a label or description for the group.

File: src/forms/labels/labeled-field.js [hunk 2]
```diff
@@ -88,23 +91,25 @@ function LabeledField({
...
   const { touched, invalid } = meta
   return (
-    <fieldset
-      className={classnames(className, {
+    <Component
+      className={classnames(className, 'field-wrapper', {
         error: hasInputError({ touched, invalid }),
         disabled: rest.disabled,
       })}
+      role="group"
```
Comment (jhp0621) [inline src/forms/labels/labeled-field.js:105]: **aria role = group** MDN: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/group_role

Does it make sense to assign `group` role to the container here? 

It is technically redundant when Component is `fieldset` whose implicit aria role is group (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset#technical_summary)

This would be for when Component is `div`:
Screen Shot 2023-09-25 at 11 39 51 AM
This role assignment would be necessary to pass the test that relies on role (if we are to keep the test as is): https://github.com/LaunchPadLab/lp-components/blob/main/test/forms/inputs/icon-input.test.js#L12

What about for individual checkbox/radio group? Is it odd that there is a `role=group` for the container that wraps the individual label and the corresponding checkbox?
Screen Shot 2023-09-25 at 11 38 41 AM

cc: @chawes13

**Review by danparnella — COMMENTED** [review]
A couple quick thoughts. Will try to add more tonight or tomorrow morning.

File: src/forms/inputs/checkbox-group.js [hunk 3]
```diff
@@ -123,30 +144,41 @@ function CheckboxGroup(props) {
       return onChange(newValueArray)
     }
   }
+
+  const CheckboxOptions = () =>
+    optionObjects.map((option, i) => {
+      return (
+        <Checkbox // eslint-disable-line react/jsx-key
```
[the remark this answers is not in the document]
Comment (chawes13) [reply src/forms/inputs/checkbox-group.js]: We should also not pass `i` for key 🙂. What property do we have on `option` that must be unique?

[the remark this answers is not in the document]
Comment (danparnella) [reply src/forms/inputs/checkbox-group.js]: Ji already updated that to `option.value`. It was in my original comment and I removed that before submitting when I saw the recent changes.

File: src/forms/inputs/checkbox-group.js [hunk 4]
```diff
@@ -34,6 +36,7 @@ import {
  * @param {Object} meta - A `redux-form` [meta](http://redux-form.com/6.5.0/docs/api/Field.md/#meta-props) object
  * @param {Array} options - An array of checkbox values (strings, numbers, or key-value pairs)
  * @param {Object} [checkboxInputProps={}] - An object of key-value pairs representing props to pass down to all checkbox inputs
+ * @param {Boolean} [useDropdown] - A boolean that determines whether the checkbox options are displayed in a dropdown container or not (optional, default `false`)
```
Comment (danparnella) [inline src/forms/inputs/checkbox-group.js:39]: I think if you update `[useDropdown]` to `[useDropdown=false]`, that will automatically add in the "(optional, default `false`)" into the generated Docs (like `checkboxInputProps` above).

File: src/forms/inputs/checkbox-group.js [hunk 5]
```diff
@@ -88,17 +91,45 @@ const propTypes = {
...
+        <span className="required-indicator" aria-hidden="true">
+          {requiredIndicator}
+        </span>
+      )}
+      {hint && <i>{hint}</i>}
+    </legend>
+  )
+}
+
+function CheckboxOptionContainer({ children, useDropdown, value }) {
```
Comment (danparnella) [inline src/forms/inputs/checkbox-group.js:124]: Nit: Should we pluralize `Option` in `CheckboxOptionContainer` since this wraps all of the options?

**Review by danparnella — CHANGES_REQUESTED** [review]
Still going through it, but here's a bit more feedback!

File: src/forms/inputs/checkbox-group.js (hunk 4 above)
Comment (chawes13) [inline src/forms/inputs/checkbox-group.js:39]: I'm pretty hesitant on the naming here. `useX` in React-land now has a very specific set of expectations with it after the release of hooks. I'd prefer to leave that prefix for custom hooks and not for prop names, if possible.

File: src/forms/inputs/radio-group.js [hunk 6]
```diff
@@ -89,9 +90,18 @@ const defaultProps = {
   radioInputProps: {},
 }
 
-function RadioGroupLegend({ label, name }) {
-  if (label === false) return null
-  return <legend>{label || convertNameToLabel(name)}</legend>
+function RadioGroupLegend({ label, name, required, requiredIndicator, hint }) {
+  return (
+    <legend className={classnames({ 'visually-hidden': label === false })}>
```
Comment (chawes13) [inline src/forms/inputs/radio-group.js:95]: I think we'll also need to add a note about this in the migration guide. If the user is coming from an older project, they might not have `visually-hidden` styles included in their scss set up (or they might have it applied by another name).

File: src/forms/labels/labeled-field.js [hunk 7]
```diff
@@ -7,9 +7,9 @@ import { hasInputError } from '../helpers'
 
 /**
  *
- * A fieldset wrapper for redux-form controlled inputs. This wrapper adds a label component (defaults to {@link InputLabel})
+ * A container wrapper for redux-form controlled inputs. This wrapper adds a label component (defaults to {@link InputLabel})
```
Comment (chawes13) [inline src/forms/labels/labeled-field.js:10]: IMO "container wrapper" is redundant. It's a wrapper or a container, but not both?

File: src/forms/labels/labeled-field.js [hunk 8]
```diff
@@ -20,6 +20,7 @@ import { hasInputError } from '../helpers'
  * @param {Boolean} [hideErrorLabel] - A boolean determining whether to hide the error label on input error (optional, default `false`)
  * @param {Function} [labelComponent=InputLabel] - A custom label component for the input
  * @param {Function} [errorComponent=InputError] - A custom error component for the input
+ * @param {Element Type} [as] - A string that determines the element type of the component (optional, default `div`)
```
Comment (chawes13) [inline src/forms/labels/labeled-field.js:23]: You can specify props as optional and their default automatically using JSDoc syntax. Refer to `labelComponent` above for how to accomplish that. While we're here, we might as well make that update to `hideErrorLabel` as well.

Comment (chawes13) [inline src/forms/labels/labeled-field.js:23]: If this prop was really a string, then the @param should be `String` right? But `propTypes.elementType` supports functional components and class components as well (https://github.com/facebook/prop-types/pull/211/files).

Do you have a source for using `elementType` for `"as"`?

File: docs.md [hunk 9]
```diff
@@ -1112,19 +1065,19 @@ function EmailInput ({
...
+use the [omitLabelProps][88] helper.
 
 ### Parameters
 
-*   `hideErrorLabel` **[Boolean][144]?** A boolean determining whether to hide the error label on input error (optional, default `false`)
-*   `labelComponent` **[Function][143]** A custom label component for the input (optional, default `InputLabel`)
-*   `errorComponent` **[Function][143]** A custom error component for the input (optional, default `InputError`)
+*   `hideErrorLabel` **[Boolean][141]?** A boolean determining whether to hide the error label on input error (optional, default `false`)
+*   `labelComponent` **[Function][140]** A custom label component for the input (optional, default `InputLabel`)
+*   `errorComponent` **[Function][140]** A custom error component for the input (optional, default `InputError`)
```
Comment (chawes13) [inline docs.md:1080]: Hmm, I would expect `as` to show up here. Maybe it's not showing up because JSDoc doesn't respect "Element Type" as a type? I don't think we can assume the naming convention in the prop types library maps 1:1 to JSDoc

File: stories/forms/inputs/checkbox-group.story.js [hunk 10]
```diff
@@ -75,3 +75,6 @@ storiesOf('CheckboxGroup', module)
       }}
     />
   ))
+  .add('with dropdown options', () => (
+    <CheckboxGroup input={inputProps} meta={{}} options={options} useDropdown />
```
Comment (chawes13) [inline stories/forms/inputs/checkbox-group.story.js:79]: Prefer to be explicit here, based on rationale presented here: https://legacy.reactjs.org/docs/jsx-in-depth.html#props-default-to-true

File: stories/forms/inputs/checkbox-group.story.js [hunk 11]
```diff
@@ -75,3 +75,6 @@ storiesOf('CheckboxGroup', module)
       }}
     />
   ))
+  .add('with dropdown options', () => (
```
Comment (chawes13) [inline stories/forms/inputs/checkbox-group.story.js:78]: The name of this story almost reads as if the options themselves have some form of dropdown capability. Could you rephrase this to be more clear?

**Review by chawes13 — COMMENTED** [review]
In addition to adding a migration guide that details the changes that a user would have to make (including updating their stylesheets), we should address the regression in test coverage. Right now on main we're very close to 100%, this PR leaves some pathways uncovered (report available via `yarn test`).

**Main**


**This branch**

File: .storybook/styles/application.scss [hunk 12]
```diff
@@ -48,10 +48,9 @@ $output-bourbon-deprecation-warnings: false;
 @import 'components/progress-circle';
 @import 'components/login';
 @import 'components/pagination';
-@import 'components/sidebar-content';
```
[the remark this answers is not in the document]
Comment (danparnella) [reply .storybook/styles/application.scss:51]: Ji was referencing my original PR where I removed a couple stylesheets that weren't being used: https://github.com/LaunchPadLab/lp-components/pull/514/files#r1132970196.

[the remark this answers is not in the document]
Comment (chawes13) [reply .storybook/styles/application.scss:51]: Thanks. Are we good to close that PR in favor of this one?

[the remark this answers is not in the document]
Comment (danparnella) [reply .storybook/styles/application.scss:51]: Closed! I'm still reviewing them side-by-side to see if there's anything else we can pull from there, but it was safe to close.

File: src/forms/inputs/radio-group.js [hunk 13]
```diff
@@ -89,9 +90,18 @@ const defaultProps = {
   radioInputProps: {},
 }
 
-function RadioGroupLegend({ label, name }) {
-  if (label === false) return null
-  return <legend>{label || convertNameToLabel(name)}</legend>
+function RadioGroupLegend({ label, name, required, requiredIndicator, hint }) {
```
Comment (danparnella) [inline src/forms/inputs/radio-group.js:93]: Since this is the same for the checkbox group, would it make sense to pull this out into it's own component (just a `legend` component)? That way it could also be used separately on projects if needed.

File: src/forms/inputs/checkbox-group.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (danparnella) [reply src/forms/inputs/checkbox-group.js]: Marking unresolved as I think we should also remove the lint disabling by pulling `key` up out of the props spread.

File: src/forms/inputs/radio-group.js [hunk 14]
```diff
@@ -137,6 +147,7 @@ function RadioGroup(props) {
     <LabeledField
       className={className}
       labelComponent={RadioGroupLegend}
+      as="fieldset"
       {...props}
     >
       {optionObjects.map((option, i) => {
```
Comment (danparnella) [inline src/forms/inputs/radio-group.js:153]: Might as well make the same changes made to [Checkbox Group](https://github.com/LaunchPadLab/lp-components/pull/595/files/036cf6d2d2be75b107bfa164dea4bb8a8c5fe274#r1334902078) here as well:
- Removing the lint disabling
- Pulling up `key` out of the props spread
- Setting `key` to `option.value` instead of index

File: src/forms/labels/labeled-field.js (hunk 2 above)
Comment (danparnella) [reply src/forms/labels/labeled-field.js:105]: Agreed. Since we are providing the ability now to set this to the correct element that would provide the correct aria role, I wouldn't think we'd need to specify one and should modify the test instead.

File: test/forms/inputs/checkbox-group.test.js [hunk 15]
```diff
@@ -125,3 +125,21 @@ test('CheckboxGroup passes down props to children', () => {
   expect(checkbox).not.toHaveClass('custom-group-class')
   expect(checkbox).toHaveClass('custom-input-class')
 })
+
+test('DropdownCheckboxGroup sets menu no longer active when clicked outside', async () => {
```
Comment (danparnella) [inline test/forms/inputs/checkbox-group.test.js:129]: Since this isn't a real component anymore, should this be "CheckboxGroup with dropdown = true" (or similar)?

File: test/forms/inputs/checkbox-group.test.js [hunk 16]
```diff
@@ -125,3 +125,21 @@ test('CheckboxGroup passes down props to children', () => {
...
+  const user = userEvent.setup()
+
+  render(<WrappedCheckboxGroup value={['1']} useDropdown />)
+
+  const select = screen.getByRole('button')
+  await user.click(select)
+
+  expect(select.nextSibling).toHaveClass('options', 'is-active')
+
+  const fieldset = screen.getAllByRole('group').at(0)
```
Comment (danparnella) [inline test/forms/inputs/checkbox-group.test.js:139]: Once we remove the `role` attribute from `LabeledField`, does this still work? I know creating it as a `fieldset` automatically gives it a role of group, but is that actually available to select by here?

File: src/forms/labels/labeled-field.js (hunk 2 above)
Comment (danparnella) [reply src/forms/labels/labeled-field.js:105]: Just a heads up, once we remove that role, several other tests will need to be updated.

**Review by danparnella — CHANGES_REQUESTED** [review]
Ok, I've gone through everything! Overall, I think this is a really good approach to solve the issue and provides a lot of flexibility. I agree with the comments Conor left and I had a few additional ones added here as well.

File: src/forms/inputs/radio-group.js (hunk 13 above)
Comment (chawes13) [reply src/forms/inputs/radio-group.js:93]: Good idea. I'm happy to pull that out into a separate issue/PR if it helps with timing (wouldn't want that to happen without new tests, documentation, etc.)

File: .storybook/styles/components/_forms.scss [hunk 17]
```diff
@@ -571,13 +553,8 @@ Custom Dropdown Select
     }
   }
 
-  fieldset.checkbox,
-  fieldset.CheckboxGroup {
-    width: 100% !important;
-
-    input {
```
[the remark this answers is not in the document]
Comment (jhp0621) [reply .storybook/styles/components/_forms.scss:578]: Yes, this HTML structure is no longer true.

Before: https://share.zight.com/p9uY6mZ7
After: https://share.zight.com/geu1mX1W

We were unnecessarily wrapping the dropdown options in a `fieldset` and each option was also wrapped in a `fieldset`.

[the remark this answers is not in the document]
Comment (jhp0621) [reply .storybook/styles/components/_forms.scss:578]: There already is a parent `fieldset` that renders the label legend

File: package.json [hunk 18]
```diff
@@ -26,8 +26,7 @@
     "docs": "documentation build src/index.js -f md -o docs.md",
     "format": "prettier --write \"(src|test|stories|.storybook)/**/*.+(js|jsx|json|scss)\"",
     "lint": "eslint src --max-warnings=0",
-    "prepare": "husky install",
-    "prepublishOnly": "yarn run clean && yarn run build",
+    "prepare": "husky install && yarn run clean && yarn run build",
```
[the remark this answers is not in the document]
Comment (jhp0621) [reply package.json:29]: Yup 100%.

While working on figuring out how to add a private github url as a package dependency, I discovered that having babel compilation tasks under `prepublish` script does not work well if someone were to consume the repository directly via git protocol (cc: https://yarnpkg.com/advanced/lifecycle-scripts#prepublish). Although I image this use case would be rare, I still think it's good practice to place yarn build logic as part of prepare.

File: migration-guides/v.10.0.0.md [hunk 19]
```diff
@@ -0,0 +1,53 @@
+# v10.0.0 Migration Guide
+
+This version contains the following breaking changes:
+
+1. The `DropdownCheckboxGroup` component is now `CheckboxGroup` with prop `dropdown`=`true`
+2. The `CheckboxGroup` and `RadioGroup` legends now rely on `visually-hidden` class styles to hide the label from the view
+
```
Comment (chawes13) [inline migration-guides/v.10.0.0.md:7]: I think we're missing a big breaking change here – `fieldset` no longer wraps each input! This will affect base styles in `_forms.scss`, as well as any **custom** rules that rely on this structure

File: migration-guides/v.10.0.0.md [hunk 20]
```diff
@@ -0,0 +1,53 @@
...
+<DropdownCheckboxGroup input={inputProps} meta={{}} options={options} />
+
+// After:
+import { CheckboxGroup } from 'lp-components'
+
+<CheckboxGroup input={inputProps} meta={{}} options={options} dropdown={true} />
+```
+
+## 2. The `CheckboxGroup` and `RadioGroup` legends now rely on `visually-hidden` class styles to hide the label from the view. 
+To follow the accessibility guideline of `fieldset` having one, unique `legend` as its direct child, changes have been made in `CheckboxGroupLegend` and `RadioGroupLegend` to always render the legend in HTML and if `label=false` is passed into the `CheckboxGroup` and `RadioGroup` components, the class name "visually-hidden" is added to the legend in order to visually hide it. Please make sure the following class styles are included in your project's root stylesheet:
```
Comment (chawes13) [inline migration-guides/v.10.0.0.md:39]: `CheckboxGroupLegend` and `RadioGroupLegend` are not part of the public API (i.e., they aren't exported from this package). Referencing them directly may lead to confusion for developers. To mitigate this, I think you could just say `CheckboxGroup` and `RadioGroup`.

File: src/forms/helpers/dropdown-select.js [hunk 21]
```diff
@@ -15,7 +15,7 @@ const defaultProps = {
   selectedValues: [],
 }
 
-// Wraps the `DropdownCheckboxGroup` component
+// Wraps the `CheckboxGroup` component: this is only when dropdown is true.
```
Comment (chawes13) [inline src/forms/helpers/dropdown-select.js:18]: 🍔 I think this would read more naturally as "Wraps the `CheckboxGroup` component when `dropdown` is set to true."

File: src/forms/inputs/checkbox-group.js [hunk 22]
```diff
@@ -34,6 +36,7 @@ import {
  * @param {Object} meta - A `redux-form` [meta](http://redux-form.com/6.5.0/docs/api/Field.md/#meta-props) object
  * @param {Array} options - An array of checkbox values (strings, numbers, or key-value pairs)
  * @param {Object} [checkboxInputProps={}] - An object of key-value pairs representing props to pass down to all checkbox inputs
+ * @param {Boolean} [dropdown=false] - A boolean that determines whether the checkbox options are displayed in a dropdown container or not
```
Comment (chawes13) [inline src/forms/inputs/checkbox-group.js:39]: 🍔 "{Boolean} - A boolean..." is redundant. That being said, we're pretty inconsistent in this repo about that!

Here's how it looks in the documentation:
Image 2023-11-02 at 9 16 31 AM

File: src/forms/labels/labeled-field.js [hunk 23]
```diff
@@ -72,10 +73,12 @@ const propTypes = {
   ...InputError.propTypes,
   children: PropTypes.node,
   hideErrorLabel: PropTypes.bool,
+  as: PropTypes.string,
```
Comment (chawes13) [inline src/forms/labels/labeled-field.js:76]: Do we want to be this permissive or are `div` and `fieldset` the only viable options?

File: src/forms/labels/labeled-field.js [hunk 24]
```diff
@@ -88,23 +91,25 @@ function LabeledField({
...
   const { touched, invalid } = meta
   return (
-    <fieldset
-      className={classnames(className, {
+    <Wrapper
+      className={classnames(className, 'field-wrapper', {
         error: hasInputError({ touched, invalid }),
         disabled: rest.disabled,
       })}
+      data-testid="field-wrapper"
```
Comment (chawes13) [inline src/forms/labels/labeled-field.js:105]: Should this be dynamic based on the input's name? If you have a test with multiple inputs, it might be challenging to locate a particular container. Not sure there's a use case for that though 🤷

File: test/forms/inputs/checkbox-group.test.js [hunk 25]
```diff
@@ -14,7 +17,7 @@ const WrappedCheckboxGroup = (props) => {
 
   const defaultProps = {
     input: {
-      name: 'test',
+      name: name,
```
Comment (chawes13) [inline test/forms/inputs/checkbox-group.test.js:20]: 🍔 shorthand assignment would work here

File: test/forms/inputs/checkbox-group.test.js [hunk 26]
```diff
@@ -118,10 +121,86 @@ test('CheckboxGroup passes down props to children', () => {
...
+
+  const fieldset = screen.getAllByRole('group').at(0)
+
+  user.click(fieldset)
+  await waitFor(() => {
+    expect(select.nextSibling).not.toHaveClass('is-active')
+  })
+})
+
+test('when no custom required indicator provided, do not show required indicator', () => {
```
Comment (chawes13) [inline test/forms/inputs/checkbox-group.test.js:188]: There is no outer `describe` block in this test file, so this test case (and the remaining) should be prefixed with `CheckboxGroup`

File: test/forms/inputs/icon-input.test.js [hunk 27]
```diff
@@ -9,9 +9,9 @@ const props = { input: { name, value, onChange }, meta: {}, icon: 'foo' }
 
 test('IconInput adds class "icon-label" to surrounding fieldset', () => {
```
Comment (chawes13) [inline test/forms/inputs/icon-input.test.js:10]: I believe the test description needs to be rewritten not to directly reference `fieldset`

File: test/forms/labels/labeled-field.test.js [hunk 28]
```diff
@@ -10,9 +10,9 @@ test('wraps children in fieldset', () => {
       <Wrapped />
```
Comment (chawes13) [inline test/forms/labels/labeled-field.test.js:10]: I believe the test description needs to be rewritten to not directly reference `fieldset`

Comment (chawes13) [reply test/forms/labels/labeled-field.test.js:10]: Could we add a test that explicitly tests the `as="fieldset"` case? E.g.,

`expect(screen.getByRole('group')).toBeInTheDocument()`

File: test/forms/inputs/radio-group.test.js [hunk 29]
```diff
@@ -131,3 +142,58 @@ test('RadioGroup passes down props to children', () => {
     expect(el).toHaveAttribute('data-test', 'true')
   })
 })
+
+test('when no custom required indicator provided, do not show required indicator', () => {
```
Comment (chawes13) [inline test/forms/inputs/radio-group.test.js:146]: Can you prefix this test (and the following) with `RadioGroup` to match the rest of the descriptions in this file?

File: test/forms/inputs/radio-group.test.js [hunk 30]
```diff
@@ -80,11 +86,12 @@ test("RadioGroup does not have a legend when label is `false`", () => {
     label: false,
   }
   render(<RadioGroup {...props} />)
-  expect(screen.queryByRole('group', { name: 'Same Name' })).not.toBeInTheDocument()
+  expect(
+    screen.queryByRole('group', { name: formattedName })
+  ).toBeInTheDocument()
```
Comment (chawes13) [inline test/forms/inputs/radio-group.test.js:91]: Can we add an assertion about `visually-hidden` being applied?

File: test/forms/labels/labeled-field.test.js (hunk 28 above)
Comment (jhp0621) [reply test/forms/labels/labeled-field.test.js:10]: @chawes13 I'm actually not sure what this first test is accomplishing... I'm merely looking for an element with the test-id I assigned and testing that it exists (without 'error' class). Should we keep it?

Comment (chawes13) [reply test/forms/labels/labeled-field.test.js:10]: Yes we should keep it. We have several components **and** styling that rely on LabeledField returning a containing element (div or fieldset), as opposed to a fragment

Comment (chawes13) [reply test/forms/labels/labeled-field.test.js:10]: Now that you're adding `'field-wrapper'` to this container, that'd be a useful thing to assert on (especially since our styles depend on it!)

File: src/forms/labels/labeled-field.js (hunk 24 above)
Comment (jhp0621) [reply src/forms/labels/labeled-field.js]: decided to ditch `data-testid` and instead use `{ container }` from `render` results as you suggested! https://testing-library.com/docs/react-testing-library/api/#container-1

File: src/forms/labels/labeled-field.js (hunk 23 above)
Comment (jhp0621) [reply src/forms/labels/labeled-field.js]: I think we should be strict with what's allowed and if we ever receive requests re: expanding the options (for whatever reason that I can't think of atm), we can address then!

Comment (jhp0621) [reply src/forms/labels/labeled-field.js]: cc: @danparnella

File: migration-guides/v.10.0.0.md (hunk 19 above)
Comment (jhp0621) [reply migration-guides/v.10.0.0.md:9]: I also added an additional call out about `LabeledField` itself (bullet point 2) since that is an exported component that can be used on its own.. thoughts?

File: src/forms/inputs/checkbox-group.js [hunk 31]
```diff
@@ -118,22 +150,28 @@ function CheckboxGroup(props) {
     return function (checked) {
       // Add or remove option value from array of values, depending on whether it's checked
       const newValueArray = checked
-        ? addToArray([option.value], value)
+        ? addToArray(value, [option.value])
```
Comment (jhp0621) [inline src/forms/inputs/checkbox-group.js:153]: Flipped the order so that the new selection shows at the end of the input, rather than at the beginning

File: src/forms/helpers/dropdown-select.js [hunk 32]
```diff
@@ -8,27 +8,38 @@ const propTypes = {
...
-function DropdownSelect({ children, className, selectedValues }) {
+function DropdownSelect({ children, className, selectedValues, options }) {
   const [expanded, toggleExpanded] = useToggle()
 
   return (
     <OutsideClickHandler onOutsideClick={() => toggleExpanded(false)}>
-      <div className="dropdown-select">
+      <div
+        className={classnames('dropdown-select', {
+          'is-active': expanded,
```
Comment (jhp0621) [inline src/forms/helpers/dropdown-select.js:34]: It seems to be a good idea to put "is-active" class name at the top level for styling purpose (e.g., coloring the border of the input when the dropdown menu is open)

Comment (jhp0621) [reply src/forms/helpers/dropdown-select.js:34]: cc: https://github.com/LaunchPadLab/client-template/pull/456/files#diff-102f9f47138cbbdf4cafcec5b8d376a8a3e508d085c7a18668ca5704039fc207R271

File: test/forms/inputs/checkbox-group.test.js [hunk 33]
```diff
@@ -118,10 +121,87 @@ test('CheckboxGroup passes down props to children', () => {
...
-  const checkboxGroup = screen.getByRole('group', { name: 'Test Group' })
+  const checkboxGroup = screen.getByRole('group', { name: formattedName })
   const checkbox = screen.getByRole('checkbox')
 
   expect(checkboxGroup).toHaveClass('custom-group-class')
   expect(checkbox).not.toHaveClass('custom-group-class')
   expect(checkbox).toHaveClass('custom-input-class')
 })
+
+test('CheckboxGroup with dropdown = true adds value to array when unselected option clicked', async () => {
```
Comment (danparnella) [inline test/forms/inputs/checkbox-group.test.js:132]: Looks like Travis is failing on this test

**Review by danparnella — CHANGES_REQUESTED** [review]
I'll try to do one more holistic review, but I had a few comments on the changes since my last review.

File: src/forms/inputs/checkbox-group.js [hunk 34]
```diff
@@ -145,8 +183,8 @@ function CheckboxGroup(props) {
               ...checkboxInputProps,
```
Comment (danparnella) [inline src/forms/inputs/checkbox-group.js:183]: Not a new issue, but can you resolve the accessibility issue for the "with input props specified" story? Looks like we'll need to set the `id` here like we're doing in `RadioGroup` ([line](https://github.com/LaunchPadLab/lp-components/pull/595/files#diff-5c2fdf596caab0ddfe40eef9596137b37bd03591ab1496e366d363d9ad2e9ac2R164)).

File: src/forms/labels/labeled-field.js [hunk 35]
```diff
@@ -72,10 +73,12 @@ const propTypes = {
...
   hideErrorLabel: PropTypes.bool,
+  as: PropTypes.oneOf(['div', 'fieldset']),
 }
 
 const defaultProps = {
   hideErrorLabel: false,
+  as: 'div',
 }
 
 function LabeledField({
```
Comment (danparnella) [inline src/forms/labels/labeled-field.js:84]: Again, not new, but there are a few stories for this component with accessibility issues that can be seen in the storybook (form elements not having labels).

**Review by danparnella — CHANGES_REQUESTED** [review]
I found a few storybook accessibility violations for the components we've touched that we should be able to resolve pretty quickly.

File: src/forms/inputs/checkbox-group.js (hunk 34 above)
Comment (chawes13) [reply src/forms/inputs/checkbox-group.js:183]: IMO let's save this for a fast-follow patch release. I'd prefer not to delay this release any longer

File: src/forms/inputs/checkbox-group.js (hunk 31 above)
Comment (chawes13) [reply src/forms/inputs/checkbox-group.js:153]: Good catch! We should add a note about this in the migration guide

File: src/forms/labels/labeled-field.js (hunk 35 above)
Comment (danparnella) [reply src/forms/labels/labeled-field.js:84]: 👍 Issue created: https://github.com/LaunchPadLab/lp-components/issues/632

**Comment by chawes13** [comment]
Removed my review to take it out of my queue, please retag me when y'all are comfortable with the PR and ready for me to take a look!

**Review by danparnella — APPROVED** [review]
Updates resolved everything I saw previously. LGTM! Just need to resolve the version number conflict.

@chawes13 Ready for you to double-check!

File: src/forms/helpers/dropdown-select.js [hunk 36]
```diff
@@ -46,8 +57,9 @@ DropdownSelect.propTypes = propTypes
 
 DropdownSelect.defaultProps = defaultProps
 
-function getLabel(values) {
-  return values.length ? values.join(', ') : 'None'
+function getLabel(values, options) {
+  const labels = values.map((v) => options.find((o) => o.value === v).key)
+  return values.length ? labels.join(', ') : 'None'
```
Comment (chawes13) [inline src/forms/helpers/dropdown-select.js:62]: While `labels.length` should always equal `values.length`, it's safer to have the guard on `labels.length`

**Review by chawes13 — APPROVED** [review]
Lovely 👏 

Confirmed test coverage remains at 100%! 🚀 
Image 2024-02-12 at 12 01 05 PM
